module/builder/consensus/builder.go

```diff
@@ -109,6 +113,10 @@ func (b *Builder) BuildOn(parentID flow.Identifier, setter func(*flow.Header) er
 		return nil, fmt.Errorf("could not insert receipts: %w", err)
 	}
 
+	if len(insertableReceipts) > DefaultPayloadMaximumReceipts {
+		insertableReceipts = insertableReceipts[:DefaultPayloadMaximumReceipts]
```

Comment:
Nice 👍 

I would suggest to move this into `getInsertableReceipts`. The reason is that the insertableReceipts are ordered, adding it here would imply that we "know" about the order and how to pick the first X receipts. But I think we could let `getInsertableReceipts` to decide everything, including the order as well as what are the first X receipts:

```
	insertableReceipts, err := b.getInsertableReceipts(parentID, DefaultPayloadMaximumReceipts)
```

Reply:
I was also thinking about it, but it kinda makes sense to have a method which just returns all insertable seals and then filter it, looks kinda cleaner to me and might be useful.